NitroxPatcher/Patches/Dynamic/CreatureDeath_OnKillAsync_Patch.cs

```diff
@@ -44,14 +46,13 @@ public sealed partial class CreatureDeath_OnKillAsync_Patch : NitroxPatch, IDyna
     public static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions)
     {
         // First injection
-        return new CodeMatcher(instructions).MatchEndForward([
+        return new CodeMatcher(instructions).MatchStartForward([
                                                 new CodeMatch(OpCodes.Ldarg_0),
                                                 new CodeMatch(OpCodes.Ldnull),
                                                 new CodeMatch(OpCodes.Stfld),
                                                 new CodeMatch(OpCodes.Br),
                                             ])
-                                            .Advance(1)
-                                            .InsertAndAdvance(new CodeInstruction(OpCodes.Ldarg_0))
+                                            .InsertAndAdvance(new CodeInstruction(OpCodes.Ldloc_1))
```

Comment:
This is my bad, in IEnumerators the current class reference "this" is accessed by Ldloc_1 and not Ldarg_0 (like in a regular function)